### PR TITLE
✨Add Didomi to the CMPs list in the amp-consent extension

### PR DESCRIPTION
**Pull request by MaximePlancke**
As discussed on this thread https://github.com/ampproject/amphtml/issues/17742, AMP will include CMP integration on his extension `amp-consent`. This PR integrate Didomi to the list of CMPs compatible with AMP.

You can find an example of integration here: https://sdk-amp.privacy-center.org/demo.html
If you don't see the banner, please run this in your console : `AMP.toggleExperiment('amp-consent-v2')`. When you click on accept, you will see appearing the picture, originally blocked before a consent was given.

Limitations and improvements:
- We are currently only supporting and sharing the IAB `consentString` and returning `accepted` as the `consentState` value for every consent given. Indeed, Didomi is not working with a global consent but a **consent by vendors**. We need to take care of the custom vendors/purposes globally in our SDK to support the `consentState` correctly.
- We are waiting on the changes from AMP discussed on https://github.com/ampproject/amphtml/issues/17742 to implement a better way to display, or not, the banner from the response of the POST endpoint (e.g: add the consent string in the parameters and call the endpoint every time to determine if the notice needs to be display or not)

Not sure what to reference. Is it the one ? https://github.com/ampproject/wg-ads/issues/1

### CODE REVIEW

**Comment by torch2424** [comment]
@MaximePlancke Thank you for this!

Let me know if you need help with CLA, or if you have a corporate CLA signed.

Thanks! 😄

**Comment by zhouyx** [comment]
@MaximePlancke Thank you for the work.
Could you please upload an example page and documentation file. I'm taking vacation, @torch2424 please help with reviewing. Thank you

**Comment by MaximePlancke** [comment]
Hello @torch2424 @zhouyx 

Thank you for your replies. We have just signed a CLA for Didomi. It is linked to my Didomi email address. I will push a new commit with it. 

Concerning the example + doc, we have an example here: https://sdk-amp.privacy-center.org/demo.html What exactly do you want as a doc? We are going to update our documentation on how to use Didomi as a CMP with AMP for our clients but in this example it is already configured.

Thank you

**Comment by MaximePlancke** [comment]
Hello @torch2424 

Seems like it didn't work. If you could help me with that that would be great. The email address is maxime.plancke@didomi.io

Thank you in advance :)

**Comment by torch2424** [comment]
cc @mrjoro for help with the corporate CLA 😄 

For the example / documentation, I think @zhouyx is referring to something we do similar for Ads. Which is something I'll go ahead and whip up now while we sort out your CLA.

Thanks! 👍

**Comment by mrjoro** [comment]
@MaximePlancke and @torch2424, I don't yet see Didomi in the list of corporate CLAs, though since it is a manual process to approve corporate CLAs it can take a few days.  Maxime did Didomi just sign the corp CLA today?

File: extensions/amp-consent/0.1/cmps.js [hunk 1]
```diff
@@ -33,4 +33,10 @@ if (getMode().test || getMode().localDev) {
     'checkConsentHref': '/get-consent-v1',
     'promptUISrc': '/test/manual/diy-consent.html',
   };
+
+  CMP_CONFIG['didomi'] = {
+    'consentInstanceId': 'didomi',
+    'checkConsentHref': 'https://api.privacy-center.org/amp/check-consent',
+    'promptUISrc': 'https://sdk-amp.privacy-center.org/loader.html',
+  };
 }
```
Comment (torch2424) [inline extensions/amp-consent/0.1/cmps.js]: Also, I just noticed, you wrapped your CMP in the `if (getMode().test || getMode().localDev) {`. 

I suspect you want your CMP to be available outside of testing and development mode correct? As in, you want PROD users to be able to use your CMP? Could you move this out of the `if` block?

Thanks! 😄

Comment (MaximePlancke) [reply extensions/amp-consent/0.1/cmps.js]: @torch2424 sure, you're right. I'll do the changes. 

Thank you :)

**Comment by MaximePlancke** [comment]
@mrjoro Indeed we just sign it today. I guess I'll try again in a few days :) 

Thank you

**Comment by torch2424** [comment]
@MaximePlancke 

After #21752 is merged, do you mind if I ask you add yourself to our own CMP documentation and to provide an example for our new cmp-vendors example page? The plan would to also have a list of supported vendors on the AMP documentation page so users could find your vendor from our docs 😄 

cc @zhouyx

**Comment by MaximePlancke** [comment]
@torch2424 

Sure there is no problem. Could you provide more information about what we will have to do? Where in the doc and where in the example folder do we have to add our CMP?

Thank you :)

**Comment by torch2424** [comment]
@MaximePlancke 

Awesome thank you! So yes, once we merge #21752 I will pass that info along. The reason why I don't do that now, as a lot of the team is currently travelling or working towards the upcoming AMP Conf. Once we get back and sync from that, I'll be sure to ping you again. For now, I think we will be fine by just having your users using the documentation provided on your site.

Thanks! 😄

**Comment by MaximePlancke** [comment]
@torch2424 

Alright no problem :). Thank you for the explanation

**Comment by zhouyx** [comment]
@MaximePlancke 

> We decide that the current consentState value is confusing because it is accepted/rejected/unknown in some cases, but true/false/null in other cases. To fix this and also to be backward compatible we decide to introduce new consentStateValue which always resolve to accepted/rejected/unknown

We are going to deprecate the usage of `consentState` which pass consent state value to iframe in the name attribute. Instead we will be changing the name to `consentStateValue`. Please see the updated doc change #22308 

Could you please review your implementation and change the usage. We will be deprecating the `consentState` pretty soon. Thank you very much! Please let me know if there's any question.

**Comment by MaximePlancke** [comment]
Hi @zhouyx 

Thank you for the notification. We will spend some time next week to make those changes.

Thank you!

**Comment by MaximePlancke** [comment]
Hi @zhouyx 

The changes has been made on our side. Thank you :)

**Pull request merged**